File: sorting/bubble.rb
module Bubble
  extend self

  def sort(arr)
    loop do
      swapped = false

      for index in 0..(arr.length - 2) do
        if arr[index] > arr[index + 1]
          arr[index], arr[index+1] = arr[index+1], arr[index]
          swapped = true
        end
      end

      return arr unless swapped
    end
  end
end

File: sorting/bubble.py
def sort(arr):
  swapped = True

  while swapped:
    swapped = False

    for i in range(len(arr) - 1):
      if arr[i] > arr[i + 1]:
        arr[i], arr[i + 1] = arr[i +1], arr[i]
        swapped = True

  return arr
